drivers/docker/utils.go

```diff
@@ -21,32 +22,32 @@ import (
 	"github.com/docker/docker/registry"
 )
 
-func parseDockerImage(image string) (repo, tag string) {
-	// decode the image tag
-	splitted := strings.SplitN(image, "@", 2)
-	repoTag := splitted[0]
-	idx := strings.LastIndex(repoTag, ":")
-	if idx < 0 {
-		repo = repoTag
-	} else if t := repoTag[idx+1:]; !strings.Contains(t, "/") {
-		repo = repoTag[:idx]
-		tag = t
-	} else if t := repoTag[idx+1:]; strings.Contains(t, "/") {
-		repo = image
-		tag = "latest"
-	}
+var (
+	NoPathInImageErr = errors.New("no repo found on imageID")
```

Comment:
This error is a little more specific than we can assert -- the problem could be the repo, or the tag, or the digest. Which might be confusing when it gets bubbled up to the user in the task event. So let's make this more generic:

```suggestion
	NoPathInImageErr = errors.New("does not match registry specification")
```

Then we can wrap this error in the callers to include the ID that's actually being parsed (I'll leave suggestions for those).